odoo/model/invoice_category.go

```diff
@@ -0,0 +1,82 @@
+package model
+
+import (
+	"context"
+
+	"github.com/vshn/appuio-odoo-adapter/odoo"
+)
+
+// InvoiceCategory (alias "Section" in Invoices) visually categorizes line items into logical groups.
+type InvoiceCategory struct {
+	// ID is the data record identifier.
+	ID        int    `json:"id,omitempty"`
+	Name      string `json:"name,omitempty"`
+	Sequence  int    `json:"sequence,omitempty"`
+	PageBreak bool   `json:"pagebreak,omitempty"`
+	Separator bool   `json:"separator,omitempty"`
+	SubTotal  bool   `json:"subtotal,omitempty"`
+}
+
+// CreateInvoiceCategory creates a new invoice category and returns the ID of the data record.
+// Note that setting InvoiceCategory.ID in the payload doesn't have an effect.
+func (o Odoo) CreateInvoiceCategory(ctx context.Context, category InvoiceCategory) (int, error) {
+	return o.client.CreateGenericModel(ctx, o.session, odoo.NewCreateModel("sale_layout.category", category))
+}
+
+// UpdateInvoiceCategory updates a given invoice category and returns true if the data record has been updated.
+func (o Odoo) UpdateInvoiceCategory(ctx context.Context, category InvoiceCategory) (bool, error) {
```

Comment:
Question towards every method in this file: Should the context be passed in every time or passed in via constructor to have it reusable as a field? the latter would simplify the signature a bit.

Reply:
I prefer passing it every time. The standard library does it this way [1](https://pkg.go.dev/database/sql#Conn.ExecContext). `http.Request` [2](https://pkg.go.dev/database/sql#Conn.ExecContext) is different but similar too, the `Request` is consumed by `Do`, this makes the context passed to the constructor equally short lived.

The big advantage I see is cancellations/timeouts per request.

Reply:
IMO you should never :tm:  put a context in a struct. We should pass it separately to every call. If we put the context in the `Odoo` struct we lose most of the features of the context. Most importantly the user of the library is now unable cancel individual calls.

Reply:
ack, keeping it per method :+1: